### PR TITLE
Changes usage of _ASSERT for SUCCESS error checking and _ASSERT with and conditional

### PR DESCRIPTION
**Pull request by Gvilla1000-nasa**
## Description
Adds new macro for `_ASSERT` when it is used specifically for checks against SUCCESS. The new macro is called `_ASSERT_SUCCESS` and compares two inputs against a success return code (0). 

Additionally, this PR adds a macro for when `_ASSERT ` is used with conditional `.and.` through the use of `_ASSERTS` (which can be renamed to another thing if preferred). 

My original intention with this PR was to create a macro for conditional `.and.` and conditional `.or.` but particularly with `.or.` there are multiple instances with varying number of arguments. I tried experimenting with having optional arguments but was unsuccessful. Because of this, the implementation of the `_ASSERTS` one is limited in usage to only two arguments so perhaps it might be better to just leave the `_ASSERT_SUCCESS` in this PR and remove the other macro?

## Motivation and Context

Usage of a specific _ASSERT case can be converted to a macro.

## How Has This Been Tested?

Tested in intel by changing the error code from good to bad to ensure macro worked as expected. When bad, error message was provided as expected. When good, result was zero diff.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [x] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Review by tclune — COMMENTED** [review]
- For `_ASSERTS` wouldn't it be clearer to implement it as calling `_ASSERT` with an `.and.`?
- Am thinking `_MASSERT` or `_M_ASSERT` might be better name.
- `_ASSERT_SUCCESS`  seems like could just be `_ASSERTS` - is the intent just to make it a bit shorter on the line? (Not even sure it does that.)

File: Tests/ExtDataDriverGridComp.F90 [hunk 1]
```diff
@@ -545,7 +545,7 @@ subroutine run(this, rc)
     integer :: userRc
 
     call ESMF_GridCompRun(this%gc, userRC=userRC,rc=status)
-    _ASSERT(userRC==ESMF_SUCCESS .and. STATUS==ESMF_SUCCESS,'run failed')
+    _ASSERT_SUCCESS(userRC,STATUS,'run failed')
```
Comment (tclune) [inline Tests/ExtDataDriverGridComp.F90:548]:
```suggestion
  call ESMF_GridCompRun(this%gc, userRC=userRC,__RC__)
 _VERIFY(userRC)
```

File: base/MAPL_CubedSphereGridFactory.F90 [hunk 2]
```diff
@@ -637,7 +637,7 @@ subroutine check_and_fill_consistency(this, unusable, rc)
       if ( (this%target_lon /= MAPL_UNDEFINED_REAL) .and. &
            (this%target_lat /= MAPL_UNDEFINED_REAL) .and. &
            (this%stretch_factor /= MAPL_UNDEFINED_REAL) ) then
-         _ASSERT( (this%target_lat >= -90.0) .and. (this%target_lat <= 90), 'latitude out of range')
+         _ASSERTS(this%target_lat >= -90.0, this%target_lat <= 90, 'latitude out of range')
```
Comment (tclune) [inline base/MAPL_CubedSphereGridFactory.F90:640]:
```suggestion
         _M_ASSERT(this%target_lat >= -90.0, this%target_lat <= 90.0, 'latitude out of range')
```
```suggestion
         _ASSERT2(this%target_lat >= -90.0, this%target_lat <= 90.0, 'latitude out of range')
```

File: base/MAPL_Profiler.F90 [hunk 3]
```diff
@@ -348,7 +348,7 @@ subroutine MAPL_TimerModeSet(MODE, RC)
 
       ! Sanity check
       timerMode = mode
-      _ASSERT(timerMode >= MAPL_TimerModeOld .and. timerMode <= MAPL_TimerModeMinMax,'needs informative message')
+      _ASSERTS(timerMode >= MAPL_TimerModeOld, timerMode <= MAPL_TimerModeMinMax,'needs informative message')
```
Comment (tclune) [inline base/MAPL_Profiler.F90:351]:
```suggestion
      _ASSERTS(timerMode >= MAPL_TimerModeOld, timerMode <= MAPL_TimerModeMinMax,'inconsistent value for timerMode')
```

**Comment by Gvilla1000-nasa** [comment]
I believe the latest push addresses all the suggestions.

**Comment by mathomp4** [comment]
> I believe the latest push addresses all the suggestions.

@Gvilla1000-nasa I think we'd also discussed making the non-userRC `.and.` asserts into two separate asserts? Though I suppose that could be a separate PR.

**Review by tclune — APPROVED** [review]

**Comment by Gvilla1000-nasa** [comment]
> > I believe the latest push addresses all the suggestions.
> 
> @Gvilla1000-nasa I think we'd also discussed making the non-userRC `.and.` asserts into two separate asserts? Though I suppose that could be a separate PR.

Working on it right now. Found a missing .and. with error checking as well

File: base/MAPL_sun_uc.F90 [hunk 4]
```diff
@@ -2994,7 +2994,8 @@ subroutine MAPL_SunGetDaylightDurationMax(ORBIT,LATS,DAYL,currTime,CLOCK,RC)
          ! obliquity fixed in this case
          OBQ = ORBIT%OB * (MAPL_PI/180.)
       endif
-      _ASSERT(0. <= OBQ .and. OBQ < MAPL_PI, 'Strange obliquity!')
+      _ASSERT(OBQ >= 0, 'Obliquity less than 0 detected!')
```
Comment (tclune) [inline base/MAPL_sun_uc.F90:2997]: Not going to ask for changes here, but we really need to develop a consistent policy about how to word such messages.  And then _gradually_ propagate.

**Review by tclune — APPROVED** [review]

**Pull request merged**